File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.py
import sys

input = sys.stdin.readline

dwarfs = [int(input().strip()) for _ in range(9)]

total_sum = sum(dwarfs)
target_sum = 100
excess = total_sum - target_sum

idx1, idx2 = -1, -1
for i in range(8):
    for j in range(i + 1, 9):
        if dwarfs[i] + dwarfs[j] == excess:
            idx1, idx2 = i, j
            break
    if idx1 != -1:
        break

real_dwarfs = [dwarfs[i] for i in range(9) if i != idx1 and i != idx2]

real_dwarfs.sort()

sys.stdout.write('\n'.join(map(str, real_dwarfs)) + '\n')

File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        StringBuilder result = new StringBuilder();
        int sum = 0;
        int[] dwarfs = new int[9];
        ArrayList<Integer> realDwarfs = new ArrayList<>();
        for(int i = 0; i < 9; i++){
            dwarfs[i] = Integer.parseInt(br.readLine().trim());
            sum += dwarfs[i];
        }
        int diff = sum - 100;
        int idx1 = 0, idx2 = 0;

        for(int i = 0; i < 8; i++){
            for(int j = i + 1; j < 9; j++){
                if(dwarfs[i] + dwarfs[j] == diff){
                    idx1 = i;
                    idx2 = j;
                    break;
                }
            }
        }

        for(int i = 0; i < 9; i++){
            if(i == idx1 || i == idx2) continue;
            realDwarfs.add(dwarfs[i]);
        }
        
        realDwarfs.sort((a, b) -> (a > b ? 1 : -1));
        
        for(int i = 0; i < 7; i++){
            result.append(realDwarfs.get(i)).append('\n');
        }

        bw.write(String.valueOf(result));
        bw.flush();
        br.close();
        bw.close();
    }
}